### PR TITLE
fetch recent revisions on mount

### PR DESCRIPTION
**Pull request by kimberlythegeek**

### CODE REVIEW

**Comment by kimberlythegeek** [comment]
I did this in the App components instead of SearchView because of the changes in #92 

In order to use hooks (`useState`, etc), the component must be a functional component, not a class component. I could have made SearchView a class and used local `state`, but this seemed more complicated for using the hooks.

File: src/components/App.tsx [hunk 1]
```diff
@@ -4,23 +4,32 @@ import CssBaseline from '@mui/material/CssBaseline';
 import { ThemeProvider } from '@mui/material/styles';
 import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 
+import store from '../common/store';
 import protocolTheme from '../theme/protocolTheme';
+import { fetchRecentRevisions } from '../thunks/searchThunk';
```
Comment (beatrice-acasandrei) [inline src/components/App.tsx:9]: You can now add as param` hide_reviewbot_pushes=true` to endpoint call to exclude results for reviewbot

**Review by julienw — CHANGES_REQUESTED** [review]
I believe it would be better to move the fetch in another new component that renders nothing instead of adding it here.

You can see more information in the bigger component. Happy to discuss this more!

File: src/components/App.tsx [hunk 2]
```diff
@@ -4,23 +4,32 @@ import CssBaseline from '@mui/material/CssBaseline';
...
 import SearchView from './Search/SearchView';
 
-function App() {
-  return (
-    <React.Fragment key="App Component">
```
Comment (julienw) [inline src/components/App.tsx:12]: while I'm here, I wonder why there is this key, it doesn't seem useful.
Also `React.Fragment` can be shortened to `<> .. </>`

File: src/components/App.tsx [hunk 3]
```diff
@@ -4,23 +4,32 @@ import CssBaseline from '@mui/material/CssBaseline';
...
-          </Routes>
-        </Router>
-      </ThemeProvider>
-    </React.Fragment>
-  );
+class App extends React.Component {
+  componentDidMount() {
+    const repo = store.getState().search.repository;
+    store.dispatch(fetchRecentRevisions(repo));
+  }
```
[the remark this answers is not in the document]
Comment (kimberlythegeek) [reply src/components/App.tsx]: I was beginning to work on this, but there's a bit more work involved than I expected. I've filed #98 and I'll begin working on that

File: src/components/App.tsx (hunk 2 above)
Comment (kimberlythegeek) [reply src/components/App.tsx]: #99

unsure why I added the fragment but it isn't needed anymore

**Review by beatrice-acasandrei — APPROVED** [review]
LGTM

**Review by esanuandra — APPROVED** [review]

**Pull request merged**